requests/sessions.py

```diff
@@ -164,6 +164,14 @@ def resolve_redirects(self, resp, req, stream=False, timeout=None,
             proxies = self.rebuild_proxies(prepared_request, proxies)
             self.rebuild_auth(prepared_request, resp)
 
+            # Attempt to rewind consumed file-like object
+            rewindable = (req.body_position is not None and prepared_request.body is not None)
+            if rewindable and getattr(prepared_request.body, 'seek', None) is not None:
+                try:
+                    prepared_request.body.seek(req.body_position)
+                except (IOError, OSError):
+                    pass # Probably raise an exception here so we don't hang on send?
```

Comment:
Let's begin by splitting this out to a method. We have a tendency in this project to shove too many things into an inline function. Let's fight that.